pkg/apiserver/rest/asset_scan_controller.go

```diff
@@ -132,14 +138,20 @@ func (s *ServerImpl) PutAssetScansAssetScanID(ctx echo.Context, assetScanID mode
 	}
 
 	// check that an asset scan with that id exists.
-	_, err = s.dbHandler.AssetScansTable().GetAssetScan(assetScanID, models.GetAssetScansAssetScanIDParams{})
+	existingAssetScan, err := s.dbHandler.AssetScansTable().GetAssetScan(assetScanID, models.GetAssetScansAssetScanIDParams{})
 	if err != nil {
 		if errors.Is(err, gorm.ErrRecordNotFound) {
 			return sendError(ctx, http.StatusNotFound, fmt.Sprintf("asset scan was not found. assetScanID=%v: %v", assetScanID, err))
 		}
 		return sendError(ctx, http.StatusInternalServerError, fmt.Sprintf("failed to get asset scan. assetScanID=%v: %v", assetScanID, err))
 	}
 
+	// check for valid resource cleanup state transition
+	err = existingAssetScan.ResourceCleanup.UpdateState(*assetScan.ResourceCleanup.State)
```

Comment:
I do not necessary see why this would be a problem unless we do not use the  `existingAssetScan` to update the object in the database.

However I agree that could be confusing, so probably the best to create a temporary variable string the `existingAssetScan.ResourceCleanup` and performing the status update on this object so this way it does not modify the `existingAssetScan` object itself avoiding possible future bugs when someone wants to reuse the `existingAssetScan` without realizing it has been modified for validation purposes.